gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/impl/jdbc/EmbedPreparedStatement.java

```diff
@@ -1106,23 +1107,39 @@ public void addBatch() throws SQLException {
 	  // would lead to one of the set of parameters being thrown
 	  // away
   	  synchronized (getConnectionSynchronization()) {
-  			if (batchStatements == null)
-// GemStone changes BEGIN
-  			{
-  			  batchStatements = new ArrayList<Object>();
-  			}
+            if (batchStatements == null)
+              // GemStone changes BEGIN
+            {
+              batchStatements = new ArrayList<Object>();
+            }
   			/* (original code)
   				batchStatements = new Vector();
   			*/
-
           //get a clone of the parameterValueSet and save it in the vector
           //which will be used later on at the time of batch execution.
           //This way we will get a copy of the current statement's parameter
           //values rather than a pointer to the statement's parameter value
           //set which will change with every new statement in the batch.
   	  try {
-            batchStatements.add(getParms().getClone());
-  	  } catch (Throwable t) {
+            //batchStatements.add(getParms().getClone());
+            if (executeBatchInProgress == 0) {
+              batchStatements.add(getParms().getClone());
+            } else {
+              // copy into the already existing params
+              ParameterValueSet temp = (ParameterValueSet)getParms();
+              int numberOfParameters = temp.getParameterCount();
+
+              if (batchStatementCurrentIndex < batchStatements.size()) {
+                for (int j = 0; j < numberOfParameters; j++) {
+                  ((ParameterValueSet)batchStatements.get(batchStatementCurrentIndex)).getParameter(j)
+                    .setValue(temp.getParameter(j));
+                }
+              } else {
+                batchStatements.add(getParms().getClone());
+              }
+            }
+            batchStatementCurrentIndex++;
+          } catch (Throwable t) {
   	    throw TransactionResourceImpl.wrapInSQLException(t);
   	  }
           /* (original code)
```

Comment:
clearParameters clear the parameter value set in activation. Here we have copied it. 
Also batchStatementCurrentIndex++ is done for both the cases.